src/components/GridListItem/index.tsx

```diff
@@ -0,0 +1,282 @@
+import React from 'react';
+import {StyleProp, StyleSheet, ViewStyle} from 'react-native';
+import {
+  BaseComponent,
+  Colors,
+  Spacings,
+  Text,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  Typography,
+  View
+} from 'react-native-ui-lib';
+import _ from 'lodash';
+import Image, {ImageProps} from '../image';
+
+export interface GridItemProps {
+  /**
+   * Thumbnail props object for rendering a Thumbnail item
+   */
+  thumbnail?: ImageProps;
```

Comment:
I think it's worth changing the prop name as well to `imageProps`. 
While In our private uilib, `thumbnail` make sense. 
In the public uilib it might be confusing, in that case, `imageProps` will be more clear as a prop name.